src/IndexingMsg.chpl

```diff
@@ -1078,27 +843,53 @@ module IndexingMsg
-                imLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);           
-                return new MsgTuple(errorMsg,MsgType.ERROR);
+            if gX.dtype == DType.BigInt {
+                var e = toSymEntry(gX, bigint);
+                var iv = toSymEntry(gIV,int);
+                var ivMin = min reduce iv.a;
+                var ivMax = max reduce iv.a;
+                var y = toSymEntry(gY,t);
+                if ivMin < 0 {
+                    var errorMsg = "Error: %s: OOBindex %i < 0".format(pn,ivMin);
+                    imLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg); 
+                    return new MsgTuple(errorMsg,MsgType.ERROR);
+                }
+                if ivMax >= e.size {
+                    var errorMsg = "Error: %s: OOBindex %i > %i".format(pn,ivMax,e.size-1);
+                    imLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);           
+                    return new MsgTuple(errorMsg,MsgType.ERROR);
+                }
+                ref iva = iv.a;
+                ref ya = y.a;
+                // NOTE y.etype will never be real when gX.dtype is bigint, but the compiler doesn't know that
+                var tmp = if y.etype == bigint then ya else if (y.etype == bool || y.etype == real) then ya:int:bigint else ya:bigint;
+                ref ea = e.a;
+                forall (i,v) in zip(iva,tmp) with (var agg = newDstAggregator(bigint)) {
+                    agg.copy(ea[i],v);
+                }
             }
-            //[(i,v) in zip(iv.a,y.a)] e.a[i] = v;
-            ref iva = iv.a;
-            ref ya = y.a;
-            ref ea = e.a;
-            forall (i,v) in zip(iva,ya) with (var agg = newDstAggregator(t)) {
-              agg.copy(ea[i],v);
+            else {
+                var e = toSymEntry(gX,t);
+                var iv = toSymEntry(gIV,int);
+                var ivMin = min reduce iv.a;
+                var ivMax = max reduce iv.a;
+                var y = toSymEntry(gY,t);
+                if ivMin < 0 {
+                    var errorMsg = "Error: %s: OOBindex %i < 0".format(pn,ivMin);
+                    imLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg); 
+                    return new MsgTuple(errorMsg,MsgType.ERROR);
+                }
+                if ivMax >= e.size {
+                    var errorMsg = "Error: %s: OOBindex %i > %i".format(pn,ivMax,e.size-1);
+                    imLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);           
+                    return new MsgTuple(errorMsg,MsgType.ERROR);
+                }
+                ref iva = iv.a;
+                var ya = y.a;
+                ref ea = e.a;
+                forall (i,v) in zip(iva,ya) with (var agg = newDstAggregator(t)) {
+                    agg.copy(ea[i],v);
+                }
```

Comment:
I think we should try and pull as much common logic out of the if block as possible, something like
```chapel
var iv = toSymEntry(gIV,int);
var ivMin = min reduce iv.a;
var ivMax = max reduce iv.a;
var y = toSymEntry(gY,t);
if ivMin < 0 {
    var errorMsg = "Error: %s: OOBindex %i < 0".format(pn,ivMin);
    imLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg); 
    return new MsgTuple(errorMsg,MsgType.ERROR);
}
if ivMax >= gX.size {
    var errorMsg = "Error: %s: OOBindex %i > %i".format(pn,ivMax,gX.size-1);
    imLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);           
    return new MsgTuple(errorMsg,MsgType.ERROR);
}
ref iva = iv.a;
ref ya = y.a;
if gX.dtype == DType.BigInt {
    var e = toSymEntry(gX, bigint);
    // NOTE y.etype will never be real when gX.dtype is bigint, but the compiler doesn't know that
    var tmp = if y.etype == bigint then ya else if (y.etype == bool || y.etype == real) then ya:int:bigint else ya:bigint;
    ref ea = e.a;
    forall (i,v) in zip(iva,tmp) with (var agg = newDstAggregator(bigint)) {
        agg.copy(ea[i],v);
    }
}
else {
    var e = toSymEntry(gX,t);
    ref ea = e.a;
    forall (i,v) in zip(iva,ya) with (var agg = newDstAggregator(t)) {
        agg.copy(ea[i],v);
    }
}
```
And the same type of deal in the other helper functions